conn.go

```diff
@@ -570,62 +609,6 @@ func (c *Conn) takeAction(actionGeneric HandshakeAction) Alert {
 			return AlertInternalError
 		}
 
-	case SendEarlyData:
-		logf(logTypeHandshake, "%s Sending early data...", label)
-		_, err := c.Write(c.EarlyData)
-		if err != nil {
-			logf(logTypeHandshake, "%s Error writing early data: %v", label, err)
-			return AlertInternalError
-		}
-
-	case ReadPastEarlyData:
```

Comment:
I was slightly happier having this work done outside the HandshakeState, which preserves the property that the handshake states' `Next()` methods only have to deal with handshake messages.  It also seems like it saves you the extra state value `ReadPastEarlyData`; you just have the action list returned from `Negotiated.Next()` indicate whether you should read past early data or not.  What's the rationale for moving this into the `Next()` methods?